src/System.Runtime.Extensions/src/System/IO/Path.WinRT.cs

```diff
@@ -11,12 +12,18 @@ namespace System.IO
 {
     public static partial class Path
     {
-        private static byte[] CreateCryptoRandomByteArray(int byteLength)
+        private static unsafe void GetCryptoRandomBytes(byte* bytes, int byteCount)
         {
+            Debug.Assert(bytes != null);
+            Debug.Assert(byteCount >= 0);
+
             byte[] arr;
-            IBuffer buffer = CryptographicBuffer.GenerateRandom((uint)byteLength);
+            IBuffer buffer = CryptographicBuffer.GenerateRandom((uint)byteCount);
             CryptographicBuffer.CopyToByteArray(buffer, out arr);
-            return arr;
+
+            Debug.Assert(arr.Length == byteCount);
+
+            Marshal.Copy(arr, 0, new IntPtr(bytes), byteCount);
```

Comment:
I tried the following in a UWP app...

```csharp
private static unsafe void GetCryptoRandomBytes(byte* bytes, int byteCount)
{
    Debug.Assert(bytes != null);
    Debug.Assert(byteCount > 0);

    IBufferByteAccess buffer = (IBufferByteAccess)CryptographicBuffer.GenerateRandom((uint)byteCount);
    byte* pBuffer = (byte*)buffer.GetBuffer();
    for (int i = byteCount - 1; i >= 0; i--)
    {
        bytes[i] = pBuffer[i];
    }
}

[ComImport]
[Guid("905a0fef-bc53-11df-8c49-001e4fc686da")]
[InterfaceType(ComInterfaceType.InterfaceIsIUnknown)]
internal interface IBufferByteAccess
{
    IntPtr GetBuffer();
}
```

A 1,000,000 microbenchmark shows that this is 4.8x faster than what I had originally.